src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -119,7 +120,7 @@ internal static IArgument CreateArgumentOperation(ArgumentKind kind, IParameterS
                     //TODO: https://github.com/dotnet/roslyn/issues/18722
                     //      Right now, for erroneous code, we exposes all expression in place of arguments as IArgument with Parameter set to null,
                     //      so user needs to check IsInvalid first before using anything we returned. Need to implement a new interface for invalid invocation instead.
-                    if (n.HasErrors || (object)optionalParametersMethod == null)
+                    if ((object)optionalParametersMethod == null || n.HasAnyErrors || parameters.Select(p => p.Type).Any(t => t.IsErrorType()))
```

Comment:
As discussed offline, I have added the `GetUseSiteDiagnostic` check and kept the `ErrorType` check. And added comments to make it clear this doesn't cover all cases.